### PR TITLE
Add e2e test for root path

### PR DESCRIPTION
**Pull request by tanyutu**
## Explain the feature/fix
Add the e2e test for root path

## Does this PR introduce a breaking change
NO

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: e2e/api/test/index.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html class="no-js nimbus pf-m-redhat-font">
+  <head>
+    <!-- SPA_HEAD -->
+    <!--#include virtual="/init.html" -->
```
Comment (kunyan) [inline e2e/api/test/index.html:5]: Read more in [FAQ](/spaship/spaship/wiki/FAQ)
```suggestion
    <!--#include virtual="/.include/init.html" -->
```

File: e2e/api/test/index.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <!--#include virtual="/chrome/head/$locale/head.html" -->
```
Comment (kunyan) [inline e2e/api/test/index.html:6]:
```suggestion
    <!--#include virtual="/.include/chrome/head/$locale/head.html" -->
```

File: e2e/api/test/index.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        <div class="col-sm-1"></div>
+        <div id="rhsw" class="col-sm-11" ng-app="rhswApp">
+          <main>
+            <p>This is test</p>
+          </main>
+        </div>
+      </div>
+    </div>
+    <!-- SPA_FOOTER -->
+    <!--#include virtual="/chrome/footer/$locale/footer.html" -->
```
Comment (kunyan) [inline e2e/api/test/index.html:30]:
```suggestion
    <!--#include virtual="/.include/chrome/footer/$locale/footer.html" -->
```

File: e2e/api/test/index.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <!--#include virtual="/chrome/head/$locale/head.html" -->
+    <link
+      rel="stylesheet"
+      type="text/css"
+      href="https://cdn.jsdelivr.net/npm/@patternfly/patternfly@2.44.2/patternfly.min.css"
+    />
+  </head>
+  <body>
+    <!-- SPA_HEADER -->
+    <!--#include virtual="/chrome/header/$locale/header.html" -->
```
Comment (kunyan) [inline e2e/api/test/index.html:15]:
```suggestion
    <!--#include virtual="/.include/chrome/header/$locale/header.html" -->
```

File: e2e/manager/cypress/fixtures/test/index.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html class="no-js nimbus pf-m-redhat-font">
+  <head>
+    <!-- SPA_HEAD -->
+    <!--#include virtual="/init.html" -->
+    <!--#include virtual="/chrome/head/$locale/head.html" -->
```
Comment (kunyan) [inline e2e/manager/cypress/fixtures/test/index.html:6]:
```suggestion
    <!--#include virtual="/.include/chrome/head/$locale/head.html" -->
```

File: e2e/manager/cypress/fixtures/test/index.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html class="no-js nimbus pf-m-redhat-font">
+  <head>
+    <!-- SPA_HEAD -->
+    <!--#include virtual="/init.html" -->
```
Comment (kunyan) [inline e2e/manager/cypress/fixtures/test/index.html:5]:
```suggestion
    <!--#include virtual="/.include/init.html" -->
```

File: e2e/manager/cypress/fixtures/test/index.html [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    <!--#include virtual="/chrome/head/$locale/head.html" -->
+    <link
+      rel="stylesheet"
+      type="text/css"
+      href="https://cdn.jsdelivr.net/npm/@patternfly/patternfly@2.44.2/patternfly.min.css"
+    />
+  </head>
+  <body>
+    <!-- SPA_HEADER -->
+    <!--#include virtual="/chrome/header/$locale/header.html" -->
```
Comment (kunyan) [inline e2e/manager/cypress/fixtures/test/index.html:15]:
```suggestion
    <!--#include virtual="/.include/chrome/header/$locale/header.html" -->
```

File: e2e/manager/cypress/fixtures/test/index.html [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        <div class="col-sm-1"></div>
+        <div id="rhsw" class="col-sm-11" ng-app="rhswApp">
+          <main>
+            <p>This is test</p>
+          </main>
+        </div>
+      </div>
+    </div>
+    <!-- SPA_FOOTER -->
+    <!--#include virtual="/chrome/footer/$locale/footer.html" -->
```
Comment (kunyan) [inline e2e/manager/cypress/fixtures/test/index.html:30]:
```suggestion
    <!--#include virtual="/.include/chrome/footer/$locale/footer.html" -->
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
+describe("E2E happy test for SPAship manager", function () {
+  var NAME;
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:2]:
```suggestion
  let NAME;
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  // after each test will do sso logout
+  // afterEach(function () {
+  // cy.get("#userInfo").click();
+  // cy.get("#logout-button").click();
+  // });
+
+  it("API key generation -new", function () {
+    var env_list = ["QA"];
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:24]:
```suggestion
    const env_list = ["QA"];
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    });
+    cy.contains("Create API key").click();
+    cy.get(".pf-c-input-group__text").each(($div, i, $divs) => {
+      expect($divs).to.contain(env_list[i]);
+    });
+    cy.get('button[aria-label="Close"]').click();
+  });
+
+  it("API key delete", function () {
+    var key_name = "key" + NAME;
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:40]:
```suggestion
    const key_name = "key" + NAME;
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+  // after each test will do sso logout
+  // afterEach(function () {
+  // cy.get("#userInfo").click();
+  // cy.get("#logout-button").click();
+  // });
+
+  it("API key generation -new", function () {
+    var env_list = ["QA"];
+    var key_name = "key" + NAME;
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:25]:
```suggestion
    const key_name = "key" + NAME;
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    cy.contains(key_name).siblings('td[data-label="Scope"]').children("button").click();
+    cy.wait(10);
+    cy.contains("Delete").click({ force: true });
+    cy.contains("Confirm").click();
+    cy.reload();
+    cy.contains(key_name).should("not.exist");
+  });
+
+  it("new root application and router to display", function () {
+    var deploy_evn = "Dev";
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:52]:
```suggestion
    const deploy_env = "Dev";
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    cy.get("#add-application-button").click();
+    var test_name = "name" + NAME;
+    var ref = "v_qa_1.0.0";
+    cy.get("#name").type(test_name);
+    cy.get("select").select(deploy_evn);
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:57]:
```suggestion
    cy.get("select").select(deploy_env);
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+    cy.get("#upload").attachFile("test.tar");
+    cy.contains("Submit").click();
+    cy.get("@page1value").then((value) => {
+      cy.visit(value);
+      cy.contains("This is test").should("exist");
+    });
+  });
+
+  it("New a application and list it", function () {
+    var deploy_evn = "Dev";
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:69]:
```suggestion
    const deploy_env = "Dev";
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    cy.get("#add-application-button").click();
+    var test_name = "name" + NAME;
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:71]:
```suggestion
    const test_name = "name" + NAME;
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    var test_path = "/test" + "/" + NAME;
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:72]:
```suggestion
    const test_path = "/test" + "/" + NAME;
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+    var ref = "v_qa_1.0.0";
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:73]:
```suggestion
    const ref = "v_qa_1.0.0";
```

File: e2e/manager/cypress/integration/spa_spec.js [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+    cy.get("#name").type(test_name);
+    cy.get("select").select(deploy_evn);
```
Comment (kunyan) [inline e2e/manager/cypress/integration/spa_spec.js:75]:
```suggestion
    cy.get("select").select(deploy_env);
```

**Comment by Jared-Sprague** [comment]
This is awesome! Thanks @tanyutu !

File: e2e/api/test/index.html (hunk 1 above)
Comment (tanyutu) [reply e2e/api/test/index.html:5]: fixed

File: e2e/manager/cypress/integration/spa_spec.js (hunk 14 above)
Comment (tanyutu) [reply e2e/manager/cypress/integration/spa_spec.js:57]: I didn't see any difference here.

File: e2e/api/test/index.html (hunk 2 above)
Comment (tanyutu) [reply e2e/api/test/index.html]: fix

File: e2e/api/test/index.html (hunk 3 above)
Comment (tanyutu) [reply e2e/api/test/index.html]: fix

File: e2e/manager/cypress/fixtures/test/index.html (hunk 5 above)
Comment (tanyutu) [reply e2e/manager/cypress/fixtures/test/index.html]: fix

File: e2e/manager/cypress/fixtures/test/index.html (hunk 7 above)
Comment (tanyutu) [reply e2e/manager/cypress/fixtures/test/index.html]: fix

File: e2e/manager/cypress/integration/spa_spec.js (hunk 19 above)
Comment (tanyutu) [reply e2e/manager/cypress/integration/spa_spec.js]: fix

File: e2e/manager/cypress/integration/spa_spec.js (hunk 18 above)
Comment (tanyutu) [reply e2e/manager/cypress/integration/spa_spec.js]: fix

File: e2e/manager/cypress/integration/spa_spec.js (hunk 17 above)
Comment (tanyutu) [reply e2e/manager/cypress/integration/spa_spec.js]: fix

File: e2e/manager/cypress/fixtures/test/index.html [hunk 20]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html class="no-js nimbus pf-m-redhat-font">
+  <head>
+    <!-- SPA_HEAD -->
+    <!--#include virtual="/.include/init.html" -->
+    <!--#include virtual="/.include/chrome/head/$locale/header.html" -->
```
Comment (kunyan) [inline e2e/manager/cypress/fixtures/test/index.html:6]:
```suggestion
    <!--#include virtual="/.include/chrome/head/$locale/head.html" -->
```

File: e2e/api/test/index.html [hunk 21]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html class="no-js nimbus pf-m-redhat-font">
+  <head>
+    <!-- SPA_HEAD -->
+    <!--#include virtual="/.include/init.html" -->
+    <!--#include virtual="/.include/chrome/head/$locale/header.html" -->
```
Comment (kunyan) [inline e2e/api/test/index.html:6]:
```suggestion
    <!--#include virtual="/.include/chrome/head/$locale/head.html" -->
```

**Pull request merged**